lib/node_modules/@stdlib/string/base/replace-before/lib/main.js

```diff
@@ -0,0 +1,78 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2021 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var format = require( '@stdlib/string/format' );
+
+
+// MAIN //
+
+/**
+* Replaces the substring before the first occurrence of a specified search string.
+*
+* @param {string} str - input string
+* @param {string} search - search string
+* @param {string} replacement - replacement string
+* @throws {TypeError} first argument must be a string
+* @throws {TypeError} second argument must be a string
+* @throws {TypeError} third argument must be a string
+* @returns {string} string
+*
+* @example
+* var out = replaceBefore( 'beep boop', ' ', 'foo' );
+* // returns 'foo boop'
+*
+* @example
+* var out = replaceBefore( 'beep boop', 'p', 'foo' );
+* // returns 'foop boop'
+*
+* @example
+* var out = replaceBefore( 'Hello World!', '', 'foo' );
+* // returns ''
+*
+* @example
+* var out = replaceBefore( 'Hello World!', 'XYZ', 'foo' );
+* // returns 'Hello World!'
+*/
+function replaceBefore( str, search, replacement ) {
+	var idx;
+	if ( !isString( str ) ) {
+		throw new TypeError( format( 'invalid argument. First argument must be a string. Value: `%s`.', str ) );
+	}
+	if ( !isString( search ) ) {
+		throw new TypeError( format( 'invalid argument. Second argument must be a string. Value: `%s`.', search ) );
+	}
+    if ( !isString( replacement ) ) {
+        throw new TypeError( format( 'invalid argument. Third argument must be a string. Value: `%s`.', replacement ) );
+    }
+
+	idx = str.indexOf( search );
+	if ( idx === -1 ) {
+		return str;
+	}
+	return str.replace( str.substring( 0, idx ), replacement );
```

Comment:
You can probably just use string concatenation here. I.e.,

```javascript
return replacement + str.substring( 0, idx );
```

Reply:
Oh yes, let me update that. Thank you!